test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -130,7 +137,123 @@ public void IsWarmUpRequest_ReturnsExpectedValue()
+                Assert.Equal(1, logLines.Count(p => p.Contains($"Starting Host (HostId={expectedHostId}")));
+                Assert.Contains("Generating 0 job function(s)", logLines);
+
+                WebScriptHostManager.ResetStandbyMode();
+            }
+        }
+
+        [Fact]
+        public async Task StandbyMode_EndToEnd_LinuxContainer()
+        {
+            byte[] bytes = TestHelpers.GenerateKeyBytes();
+            var encryptionKey = Convert.ToBase64String(bytes);
+
+            var vars = new Dictionary<string, string>
+            {
+                { EnvironmentSettingNames.ContainerName, "TestContainer" },
+                { EnvironmentSettingNames.ContainerEncryptionKey, encryptionKey },
+                { EnvironmentSettingNames.AzureWebsiteContainerReady, null },
+                { "AzureWebEncryptionKey", "0F75CA46E7EBDD39E4CA6B074D1F9A5972B849A55F91A248" }
+            };
+            using (var env = new TestScopedEnvironmentVariable(vars))
+            {
+                var httpConfig = new HttpConfiguration();
+
+                var testRootPath = Path.Combine(Path.GetTempPath(), "StandbyModeTest_Linux");
+                await FileUtility.DeleteDirectoryAsync(testRootPath, true);
+
+                var loggerProvider = new TestLoggerProvider();
+                var loggerProviderFactory = new TestLoggerProviderFactory(loggerProvider);
+                var webHostSettings = new WebHostSettings
+                {
+                    IsSelfHost = true,
+                    LogPath = Path.Combine(testRootPath, "Logs"),
+                    SecretsPath = Path.Combine(testRootPath, "Secrets"),
+                    ScriptPath = Path.Combine(testRootPath, "WWWRoot")
+                };
+
+                var loggerFactory = new LoggerFactory();
+                loggerFactory.AddProvider(loggerProvider);
+
+                var webHostBuilder = Program.CreateWebHostBuilder()
+                    .ConfigureServices(c => {
+                        c.AddSingleton(webHostSettings)
+                        .AddSingleton<ILoggerProviderFactory>(loggerProviderFactory)
+                        .AddSingleton<ILoggerFactory>(loggerFactory);
+                    });
+
+                var httpServer = new TestServer(webHostBuilder);
+                var httpClient = httpServer.CreateClient();
+                httpClient.BaseAddress = new Uri("https://localhost/");
+
+                TestHelpers.WaitForWebHost(httpClient);
+
+                var traces = loggerProvider.GetAllLogMessages().ToArray();
+                Assert.NotNull(traces.Single(p => p.FormattedMessage.StartsWith("Starting Host (HostId=placeholder-host")));
+                Assert.NotNull(traces.Single(p => p.FormattedMessage.StartsWith("Host is in standby mode")));
+
+                // issue warmup request and verify
+                var request = new HttpRequestMessage(HttpMethod.Get, "api/warmup");
```

Comment:
Note that just like App Service placeholder mode, the runtime starts up with a dummy Warmup function which can be invoked to warm things up. TBD if we can use that in SeaBreeze. The function is anonymous - the assumption is that it can only be called internally, since the site hasn't been assigned yet so shouldn't be externally addressable.

Reply:
/cc @bala16 if we need this in seabreeze, then we'll need the frontEnd to ping pool containers every so often to keep them warm. We haven't had issues with memory page out yet, but probably because the per-vm density is not high now